\documentclass[]{article}
\usepackage[round]{natbib}

\usepackage{fullpage}
\usepackage{listings}
\usepackage{url}
\usepackage{authblk}
\usepackage{graphicx}
\usepackage{color}
\usepackage{booktabs}

\lstset{language=Python}

% local definitions
\newcommand{\comment}[1]{{\textcolor{red}{Comment: #1}}}
\newcommand{\aprcomment}[1]{{\textcolor{blue}{APR: #1}}}

% cross-reference with supplement
\usepackage{xr}
\externaldocument{supplement}

\begin{document}

\title{On the fraught inference of historical generation times}
\author[1,*]{Aaron P. Ragsdale}
\author[2]{Kevin Thornton}
\affil[1]{University of Wisconsin--Madison, Wisconsin, USA}
\affil[2]{University of California, Irvine, California, USA}
\affil[*]{email@address.edu}
\maketitle

Possible titles:

\begin{abstract}

    \citet{wang2023human} recently proposed an approach to infer the history of
    human generation times from changes in mutation profiles over time. With
    the observation that the relative proportions of different mutation types
    depend on the ages of parents, stratifying variants by the age of mutation
    allows for the inference of average paternal and maternal generation times
    at past times. Applying this approach to published allele age estimates,
    \citeauthor{wang2023human} inferred long-lasting sex differences in average
    generation times and surprisingly found that ancestral generation times of
    West African populations remained substantially higher than those of
    Eurasian populations extending tens of thousands of generations into the
    past. Here we show that these results and interpretations are driven by
    noise and biases in input data, questionable modeling and statistical
    choices, and a failure to validate their inferences using independent
    approaches for estimating allele ages. With the recent development of
    methods to reconstruct genome-wide gene genealogies, coalescence times, and
    allele ages, we caution that ...

    \aprcomment{this seems kind of aggressive\ldots just writing down some thoughts}

\end{abstract}

Recent years have seen the rapid development of methods toward the
reconstruction of genealogical structures of large cohorts
\citep{speidel2019method,wohns2022unified,hubisz2020mapping}, comprised of a
series of gene genealogies (or trees) along the genome. Reconstructed
genealogies (or informative summaries of them \citep{albers2020dating}) have
the potential to transform evolutionary and population genetic inference, as
biological and evolutionary processes impact the shape and correlation of gene
trees and the distribution of variation that arises in the lineages they
represent. Relevant to this study, the age of a variant can be estimated by
mapping that mutation to the portion of the gene tree in which it is inferred
to have occurred.

At the same time, the last few years have seen large sequencing projects that
provide ever-increasing resolution of genome biology, including direct
measurements of mutation rates and profiles
\citep{jonsson2017parental,halldorsson2019characterizing}. Through
high-coverage sequencing of multiple generations of families, \emph{de novo}
mutations can be determined as maternally or paternally inherited, and with a
large enough sample size both the number of mutations and proportions of
mutation types (e.g., A$\rightarrow$C, A$\rightarrow$G, etc.) can be correlated
with parental age and sex \citep{jonsson2017parental}. \citet{wang2023human}
combined these two sets of inferences, the estimated ages of mutations and the
paternal age- and sex-dependence of the mutation spectrum, to infer the history
of average maternal and paternal generation intervals for human populations of
diverse ancestries. In order to avoid overfitting, this approach requires
making a number of assumptions about the constancy of the mutational process
\cite{harris,dewitt} \aprcomment{and others to point out?}. For now, let's take
these as stated\ldots

\subsection*{Long-lasting differences in human generation intervals}

In applying this inference approach to multiple populations of different
continental ancestries, \citet{wang2023human} estimated that the ancestors of
European, East Asian, and South Asian populations included in the
\citet{1000genomes2015} (1KGP) had significantly reduced average generation
times compared to West African populations included in the same dataset. These
differences extended to at least 10,000 generations, the time period that was
highlighted in this study. In discussing this result, the authors state that
``the difference among populations beyond 2000 generations ago reflects
population structure in humans before their dispersal out of Africa, a
structure that is not fully captured by the 1000 Genomes AFR sample. This
implies that the simple labels of `African' and `non-African' for these
populations conceal differences in generation times that existed on our
ancestral continent.'' Indeed, a number of recent genetic studies suggest that
human population structure within Africa extending hundreds of thousands of
years into the past has shaped modern-day genetic variation
\citep{plagnol2006possible,hammer2011genetic,hsieh2016model,hey2018phylogeny,ragsdale2019models,durvasula2020recovering,lorente2019whole}.

However, in extending their analysis farther into the past,
\citet{wang2023human} find that ancestral generation intervals do not converge
until many 10s of thousands of generations into the past. With an average
generation time of 25--30 years, this corresponds to well over one million
years ago. This observation would require some portion of the ancestries of
Eurasian and West African populations to have remained isolated for many
hundreds of thousands of years, for those structured ancestral populations to
have had large differences in average generation times over the course of this
history, and for those groups to contributed substantively to different
contemporary human populations. While such a scenario of very long-lasting
isolation among ancestral popualtions is not impossible, it is not supported by
genetic \citep{ragsdale2022weakly,others} or archaeological
\citep{scerri2018did,others} evidence, which rather suggest at least periodic
connectivity of ancestral human populations.

Rather than long-lasting isolation between a large portion of ancestry of West
African and Eurasian populations (that represented large differences in
generation intervals), Eurasian--West African divergence has been estimated at
only $\approx 75$ka (thousand years ago)
\citep[e.g.,][]{pagani2015tracing,other}. Even among studies that have inferred
deeper divergences of human populations within African would place the
Eurasian--West African divergence at around 100--150ka
\citep{schlebusch2017southern}.

\aprcomment{
    If these estimated divergences represents the majority of ancestry of the
    two populations, can we do a back of the envelope calculation to see what
    difference in generation time would be needed between the hypothesized
    minority of deeply structured ancestry contributing to these pops in order
    to explain the estimated average differences in generation times? I bet it
    would be huge.
}

\aprcomment{they do say, ``note that age estimates of mutations in the very
distant past have decreased accuracy''}

Given the implausibility of the stated result, it is natural to ask what may be
causing such mis-inferences. Below we show that multiple sources of uncertainty,
namely noise and bias in allele age inference and incosistencies in trio-base
estimates of mutation profiles, confound inferences of generation times from
time series of mutation spectra.

\subsection*{Inconsistencies in the mutation spectrum over time}

Issues:
\begin{enumerate}
    \item GEVA-Atlas variant ages result in mutation spectra that change
        rapidly beyond 10,000 generates (which is likely why the weren't
        shown?)
    \item Allele age estimates between three state-of-the-art methods
        [\texttt{GEVA} \citep{albers2020dating}, Relate
        \citep{speidel2019method}, and \texttt{tsdate}
        \citep{wohns2022unified}] are only moderately correlated
        (for example, see Figure S20 in the Supplement of
        \citet{wohns2022unified}).
    \item Allele ages provided by each method results in distinct and
        unalike mutation spectrum histories
        (Figures~\ref{fig:geva-spectra}--\ref{fig:relate-spectra}).
    \item In turn, these divergent histories provide estimates of
        generation time profiles that differ completely.
\end{enumerate}

\subsection*{Discrepancies in estimated \emph{de novo} mutation rate profiles}

\begin{enumerate}
    \item \emph{De novo} mutations from the Icelandic trio study have
        a different mutation profile than the most recent age bins
        (across all three of the variant dating methods).
    \item In the most recent age bins (0 to 50 or 80 generations ago),
        the three methods estimate very similar spectra of young standing
        variation. They are closer to each other than any of them are
        to the spectrum from \citet{jonsson2017parental}.
    \item \citet{wang2023human} acknowledge this:
        \begin{quote}
            We found that the mutation spectrum from the large pedigree study
            (14) consistently differed from the variant spectrum inferred from
            the 1000 Genomes Project data, possibly because we removed
            singletons from the polymorphism dataset to reduce errors.''
        \end{quote}
    \item However, they did not test (or at least did not show a test of)
        this hypothesis that the removal of singletons drives this signal.
        I don't think it does, but want to dive into it a bit more
        thoroughly.
        \comment{Update: GEVA does not provide ages for singletons. Relate
            and tsdate do, however, so we can test this hypothesis there.}
    \item Table~\ref{tab:recent-spectra} shows mutation spectrum from the
        most recent bin in each dataset.
    \item I also don't think the approach they took is satisfying:
        \begin{quote}
            Therefore, to obtain absolute generation times for historical
            periods, we centered the observed spectra on the most recent bin,
            subtracting its difference with the average mutation spectrum
            estimated in (14) from each historical spectrum.  This has the
            effect of assuming that parental ages in the pedigreed mutation
            dataset reflect generation times in the most recent historical bin.
        \end{quote}
        And I don't know what biases this introduces. It does have the effect
        of forcing recent bins to have roughly the same inferred average
        generation times for mothers and fathers as the iceland trio data
        ($28.2$ and $32$, resp.). It's therefore not a \emph{result} that
        recent time periods match other estimates. It's a built-in assumption
        of their model
\end{enumerate}

\subsection*{Circular reasoning in modeling choices..}

\subsection*{Conclusions}

\begin{enumerate}
    \item Allele age estimates are noisy, and probably shouldn't be used
        for such detailed inferences. You'll end up fitting the noise and
        bias of each method.
    \item DNM estimates from trios have their own sets of problems. Do we
        know where the discrepancy between trio-estimated DNM spectrum and
        observations from pop-gen data come from? Probably needs to be
        sorted out.
\end{enumerate}

\break

\subsection*{Figures}

Many of these to the supplement.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/spectrum_history.geva.max_age.10000.pdf}
    \caption{
        \textbf{\texttt{GEVA}-inferred mutation spectrum history.}
    }
    \label{fig:geva-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/spectrum_history.relate.max_age.10000.pdf}
    \caption{
        \textbf{\texttt{Relate}-inferred mutation spectrum history.}
    }
    \label{fig:relate-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/spectrum_history.tsdate.max_age.10000.pdf}
    \caption{
        \textbf{\texttt{tsdate}-inferred mutation spectrum history.}
    }
    \label{fig:tsdate-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/spectrum_history.relate.max_age.10000.singletons.pdf}
    \caption{
        \textbf{\texttt{Relate}-inferred mutation spectrum history, including singletons.}
    }
    \label{fig:relate-spectra-singletons}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/spectrum_history.tsdate.max_age.10000.singletons.pdf}
    \caption{
        \textbf{\texttt{tsdate}-inferred mutation spectrum history, including singletons.}
    }
    \label{fig:tsdate-spectra-singletons}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/inferred_generation_times.geva.pdf}
    \caption{
        \textbf{\texttt{GEVA}-inferred generation time histories.}
    }
    \label{fig:geva-gen-times}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/inferred_generation_times.relate.pdf}
    \caption{
        \textbf{\texttt{Relate}-inferred generation time histories.}
    }
    \label{fig:relate-gen-times}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/inferred_generation_times.tsdate.pdf}
    \caption{
        \textbf{\texttt{tsdate}-inferred generation time histories.}
    }
    \label{fig:tsdate-gen-times}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/goodness-of-fit.geva.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \texttt{GEVA}-inferred generation times.}
    }
    \label{fig:geva-fit}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/goodness-of-fit.relate.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \texttt{Relate}-inferred generation times.}
    }
    \label{fig:relate-fit}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.8\textwidth]{../plots/goodness-of-fit.tsdate.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \texttt{tsdate}-inferred generation times.}
    }
    \label{fig:tsdate-fit}
\end{figure}


\clearpage

\subsection*{Tables}

\begin{table}[h]
    \caption{
        \label{tab:recent-spectra}
        \textbf{Mutation profiles from the past 100 generations,
        compared to Iceland trios.}
        The most recent time bin for each method included the past $\approx150$
        generations. When singletons were included (when using data from
        \texttt{tsdate} and \texttt{Relate}), the spectra of estimated
        recent standing variation were largely similar. Note that \texttt{GEVA}
        does not report ages for singletons.
        While the three methods provide similar spectra from recent mutations,
        the spectrum from the Iceland trios differs substantially.
        The similarity in the spectrum between mutations that were phased
        in \citet{jonsson2017parental} and all mutations (phased and unphased)
        are very similar to each other.
    }
    \centering
    \begin{tabular}[t]{l|cccccc}
        \toprule
        Dataset & A$\rightarrow$C & A$\rightarrow$G & A$\rightarrow$T &
            C$\rightarrow$A & C$\rightarrow$G & C$\rightarrow$T \\
        \midrule
        \texttt{GEVA} & 0.0946 & 0.3600 & 0.0886 & 0.1201 & 0.1057 & 0.2310 \\
        \texttt{tsdate} & 0.0931 & 0.3579 & 0.0899 & 0.1146 & 0.1061 & 0.2384 \\
        \texttt{tsdate} (w/singletons) & 0.0989 & 0.3598 & 0.0908 & 0.1168 & 0.1062 & 0.2275 \\
        \texttt{Relate} & 0.0991 & 0.3610 & 0.0863 & 0.1124 & 0.1038 & 0.2374 \\
        \texttt{Relate} (w/singletons) & 0.1002 & 0.3590 & 0.0921 & 0.1164 & 0.1060 & 0.2263 \\
        \midrule
        Trios (phased) & 0.1071 & 0.4100 & 0.0881 & 0.0481 & 0.0847 & 0.2620 \\
        Trios (all mutations) & 0.1080 & 0.4086 & 0.0903 & 0.0483 & 0.0839 & 0.2610 \\
        \bottomrule
    \end{tabular}
\end{table}

\clearpage

\bibliographystyle{plainnat}
\bibliography{doc}

\end{document}
