\documentclass[]{article}
\usepackage[round]{natbib}

\usepackage{fullpage}
\usepackage{listings}
\usepackage{url}
\usepackage{authblk}
\usepackage{graphicx}
\usepackage{color}
\usepackage{booktabs}
\usepackage{amsmath}

% lorem ipsum dummy text
\usepackage{lipsum}

\lstset{language=Python}

% cross-reference with main text
\usepackage{xr}
\externaldocument{doc}

% local definitions
\newcommand{\comment}[1]{{\textcolor{red}{Comment: #1}}}
\newcommand{\aprcomment}[1]{{\textcolor{blue}{APR: #1}}}

\usepackage{xspace}
\newcommand{\GEVA}{\texttt{GEVA}\xspace}
\newcommand{\tsdate}{\texttt{tsdate}\xspace}
\newcommand{\relate}{\texttt{Relate}\xspace}

\title{Supporting Information for ``[paste title here]''}
\author[1,*]{Aaron P. Ragsdale}
\author[2]{Kevin R. Thornton}
\affil[1]{University of Wisconsin--Madison, Wisconsin, USA}
\affil[2]{University of California, Irvine, California, USA}
\affil[*]{apragsdale@wisc.edu}

\begin{document}
\maketitle

\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
\renewcommand{\theequation}{S\arabic{equation}}
\setcounter{figure}{0}
\setcounter{table}{0}
\setcounter{equation}{0}

\section*{Supplemental methods}

\subsection*{Generation times needed to explain long-lasting differences between populations}

Using the reported generation intervals from \citet{wang2023human} (as shown in
Figure S4 in their supplemental material), we explored scenarios that could
lead to the observed differences between African and non-African populations
five to ten thousand generations ago, corresponding to 150-300ka. As discussed
in the main text, a 5-10 year difference in generation intervals would require
long-lasting structure. Admixture from an unidentified, diverged human lineage
has been proposed to explain observed genetic variation in African populations
\citep[e.g.,][but see
\citet{ragsdale2022weakly}]{hey2018phylogeny,durvasula2020recovering,lorente2019whole}.
In such models, a population that was isolated for hundreds of thousands of
years contributed 5--10\% ancestry to West African populations.
(Figure~\ref{fig:durvasula-model}). The remaining 90--95\% ancestry is shared
between present-day Eurasian and West African populations, and this ancestry
would have shared historical generation times. While other models of deep
population structure in Africa have been proposed, a history of strict
isolation between diverged lineages before admixture is more likely to result
in a signal of differing ancestral generation times.


\begin{figure}[h!]
    \centering
    \includegraphics{durvasula_model}
    \caption{\textbf{A model for archaic admixture within Africa.}}
    \label{fig:durvasula-model}
\end{figure}

In such a scenario, differences in inferred average historical generation times
between West African and Eurasian populations must be due to differences in
generation times between the two diverged lineages. Using the mutation model
\citet{wang2023human} inferred from Icelandic pedigree data
\citep{jonsson2017parental}, we modeled the Eurasian mutation spectrum from
this time period using paternal and maternal generation times of 20 (18--22,
from figure S4 in \citeauthor{wang2023human}) years. The West African mutation
spectrum was modeled as a mixture between this shared spectrum and the mutation
spectrum from the diverged lineage, in proportions equal to the admixture
proportions. This assumes (1) selection does not strongly influence mutation
spectrum proportions, (2) there are no demographic effects such as severe
bottlenecks that make mutation spectrum proportions unequal to admixture
proportions, and (3) the rates of mutation accumulation along each lineage are
similar. Additionally, age- and sex-dependent mutation rates from past
populations must match the mutation model from the Icelandic trio data. It is
likely that none of these assumptions perfectly hold, but these are the same
assumptions in the original inference of generation time histories.

Given the admixture proportion $f$, inferred West African-ancestral paternal
and maternal generation intervals of 28 (27--30) and 23 (22--25) years, and the
mutation model $M(p, m)$, we found generation times $p_d$ and $m_d$ in the
diverged lineage such that
\begin{align*}
    M(28, 23) & = (1-f)M(20, 20) + fM(p_d, m_d).
\end{align*}
In fitting this model with $f=0.1$, we found $p_d\approx92$ and $m_d\approx48$.
If we assume average ancestral paternal and maternal ages in the
Eurasian-shared lineage were each 22 years, $p_d\approx76$ and $m_d\approx31$. 
With Eurasian-ancestral intervals of 22 years and $f=0.2$ (much higher than
most inferences), the paternal age would still need to be over 50 years.
From this, we conclude that the generation time history inferred by
\citet{wang2023human} is incompatible with prevailing models of deep population
structure within Africa.

\subsection*{Historical mutation spectra}

We followed the filtering choices from \citet{wang2023human} in retaining
mutations with estimated ages. Namely, triplet mutation contexts associated
with a known C$\rightarrow$T mutation pulse in Europeans
\citep{harris2015evidence} and CpG sites were removed. \GEVA does not provide
allele ages for singletons, but we considered data both with and without
singletons from \tsdate- and \relate-inferred ages. Variants with allele
frequencies greater than 98\% were removed to minimize the effect of
ancestral-state misidentification.

Variants were binned by age in 100 epochs, divided such that a roughly equal
number of variants fell within each bin, as in \citet{wang2023human}. In most
cases, we considered a maximum age of 10,000 generations. Mutation profile
trajectories and generation time histories were smoothed using the
\texttt{loess\_1d} function from the Python \texttt{loess} package, with
parameters \texttt{frac=0.5} and \texttt{degree=2}.

\subsubsection*{Allele ages from \GEVA}

\subsubsection*{Allele ages from \relate}

\subsubsection*{Allele ages from \tsdate}

\bibliographystyle{genetics}
\bibliography{doc}

\clearpage

\section{Tables and figures}

\begin{table}[h]
    \centering
    \begin{tabular}[t]{l|cccccc}
        \toprule
        Dataset & A$\rightarrow$C & A$\rightarrow$G & A$\rightarrow$T &
            C$\rightarrow$A & C$\rightarrow$G & C$\rightarrow$T \\
        \midrule
        \GEVA & 0.0946 & 0.3600 & 0.0886 & 0.1201 & 0.1057 & 0.2310 \\
        \tsdate & 0.0931 & 0.3579 & 0.0899 & 0.1146 & 0.1061 & 0.2384 \\
        \tsdate (w/singletons) & 0.0989 & 0.3598 & 0.0908 & 0.1168 & 0.1062 & 0.2275 \\
        \relate & 0.0991 & 0.3610 & 0.0863 & 0.1124 & 0.1038 & 0.2374 \\
        \relate (w/singletons) & 0.1002 & 0.3590 & 0.0921 & 0.1164 & 0.1060 & 0.2263 \\
        \midrule
        Trios (phased) & 0.0953 & 0.3649 & 0.0890 & 0.0960 & 0.1216 & 0.2332 \\
        Trios (all mutations) & 0.0962 & 0.3638 & 0.0923 & 0.0951 & 0.1202 & 0.2324 \\
        \bottomrule
    \end{tabular}
    \caption{
        \label{tab:recent-spectra}
        \textbf{Mutation profiles from the past 100 generations, compared to
        Iceland trios.} The most recent time bin for each method included the
        past $\approx150$ generations. When singletons were included (when
        using data from \tsdate and \relate), the spectra of estimated recent
        standing variation were unchanged. Note that \GEVA does not report ages
        for singletons.  While the three methods provide similar spectra from
        recent mutations, the spectrum from the Iceland pedigrees differs, in
        particular for the C$\rightarrow$A and C$\rightarrow$G classes. These
        differences are up to $2\%$ of the proportion among all mutations,
        which corresponds to an under- or over-count of up to $\sim20\%$ of
        C$\rightarrow$A and C$\rightarrow$G mutations, respectively.  This
        difference remains whether the spectrum is estimated from only
        mutations that were phased in \citet{jonsson2017parental} or from all
        mutations (phased and unphased).
    }
\end{table}

\begin{table}[h]
    \centering
    \begin{tabular}[t]{l|cccccc}
        \toprule
        Dataset & A$\rightarrow$C & A$\rightarrow$G & A$\rightarrow$T &
            C$\rightarrow$A & C$\rightarrow$G & C$\rightarrow$T \\
        \midrule
        AFR (\GEVA) & 0.103 & 0.354 & 0.094 & 0.127 & 0.098 & 0.224 \\
        EAS & 0.111 & 0.341 & 0.103 & 0.131 & 0.094 & 0.220 \\
        EUR & 0.102 & 0.355 & 0.093 & 0.125 & 0.102 & 0.222 \\
        SAS & 0.095 & 0.355 & 0.090 & 0.123 & 0.099 & 0.238 \\
        \midrule
        AFR (\relate) & 0.099 & 0.356 & 0.084 & 0.116 & 0.110 & 0.236 \\
        EAS & 0.095 & 0.359 & 0.089 & 0.115 & 0.097 & 0.245 \\
        EUR & 0.100 & 0.368 & 0.085 & 0.110 & 0.102 & 0.235 \\
        SAS & 0.104 & 0.344 & 0.090 & 0.108 & 0.107 & 0.246 \\
        \midrule
        AFR (\tsdate) & 0.092 & 0.354 & 0.087 & 0.116 & 0.110 & 0.241 \\
        EAS & 0.098 & 0.356 & 0.097 & 0.112 & 0.103 & 0.233 \\
        EUR & 0.091 & 0.363 & 0.089 & 0.117 & 0.102 & 0.238 \\
        SAS & 0.091 & 0.359 & 0.088 & 0.114 & 0.107 & 0.241 \\
        \bottomrule
    \end{tabular}
    \caption{
        \label{tab:recent-spectra}
        \textbf{Mutation profiles from the past 100 generations in continental
        population groups.}
    }
\end{table}



\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.geva.max_age.10000.pdf}
    \caption{
        \textbf{\GEVA-inferred mutation spectrum history.}
    }
    \label{fig:geva-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.relate.max_age.10000.pdf}
    \caption{
        \textbf{\relate-inferred mutation spectrum history.}
    }
    \label{fig:relate-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.tsdate.max_age.10000.pdf}
    \caption{
        \textbf{\tsdate-inferred mutation spectrum history.}
    }
    \label{fig:tsdate-spectra}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.geva.max_age.80000.pdf}
    \caption{
        \textbf{\GEVA-inferred mutation spectrum history, extending to 80,000 generations.}
    }
    \label{fig:geva-spectra-80k}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.relate.max_age.10000.singletons.pdf}
    \caption{
        \textbf{\relate-inferred mutation spectrum history, including singletons.}
    }
    \label{fig:relate-spectra-singletons}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/spectrum_history.tsdate.max_age.10000.singletons.pdf}
    \caption{
        \textbf{\tsdate-inferred mutation spectrum history, including singletons.}
    }
    \label{fig:tsdate-spectra-singletons}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics{../plots/overlapping.geva.relate.pdf}
    \caption{
        \textbf{Mutation spectrum histories from mutations that were dated by
        both \GEVA and \relate.}
    }
    \label{fig:overlap-spectra}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/inferred_generation_times.DM.geva.max_age.10000.pdf}
    \caption{
        \textbf{\GEVA-inferred generation time histories.}
    }
    \label{fig:geva-gen-times}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/inferred_generation_times.DM.relate.max_age.10000.pdf}
    \caption{
        \textbf{\relate-inferred generation time histories.}
    }
    \label{fig:relate-gen-times}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/inferred_generation_times.DM.tsdate.max_age.10000.pdf}
    \caption{
        \textbf{\tsdate-inferred generation time histories.}
    }
    \label{fig:tsdate-gen-times}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/goodness-of-fit.DM.geva.max_age.10000.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \GEVA-inferred generation times.}
    }
    \label{fig:geva-fit}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/goodness-of-fit.DM.relate.max_age.10000.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \relate-inferred generation times.}
    }
    \label{fig:relate-fit}
\end{figure}


\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{../plots/goodness-of-fit.DM.tsdate.max_age.10000.pdf}
    \caption{
        \textbf{Prediction of mutation spectrum history from
        \tsdate-inferred generation times.}
    }
    \label{fig:tsdate-fit}
\end{figure}


\break

\end{document}
